src/docfx/lib/json/ValueOrObjectConverter.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Newtonsoft.Json;
+
+namespace Microsoft.Docs.Build
+{
+    internal class ValueOrObjectConverter<T1, T2> : JsonConverter where T2 : class
```

Comment:
You can dynamically get these types from `objectType` property in `ReadJson` method

```suggestion
    internal class ValueOrObjectConverter : JsonConverter
```